### ISSUE TITLE
Caution when writing code to crawl a directory -- differences between Linux and OS X

### ISSUE THREAD

**Issue by jones-michael-s** [rhodricusack/automaticanalysis#110]
Hello all,

Just a note for Linux developers -- if you are looping over the contents of a directory returned by the Matlab dir( ) function, it's common practice to start at index = 3 because this skips over the "." and ".." entries. Be aware that under OS X, there is (sometimes) another hidden file in a directory called .DS_Store (note the leading ".") that you also need to skip over. Unfortunately, .DS_Store isn't always present so you can't just do a: if (ismac) ... in your code. You need to look for .DS_Store specifically.

Long story short, if you are hardcoding directory loops to start at 3, you are asking for trouble.

Cheers,
Mike

**Comment by tiborauer** [comment]
That is why I prefer [spm_select](https://github.com/neurodebian/spm12/blob/master/spm_select.m).

**Comment by jones-michael-s** [comment]
That is probably the best practice. Alas, there are still some places in the code where I run into dir( ) and hardcoded indexes.
